File: cableTester/cableTester/io_frm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace cableTester
{
    public partial class io_frm : Form
    {
        public dio_ctrl.DigitalIOCtrl DigitalIOCtrl1;
        public io_frm()
        {
            InitializeComponent();
        }

        private void io_frm_Load(object sender, EventArgs e)
        {

        }

        public void addDIOCtrl(string niPath, string device, string model)
        {
            // create digital IO control from DAQ instrument
            this.DigitalIOCtrl1 = new dio_ctrl.DigitalIOCtrl(niPath, model, device);

            this.DigitalIOCtrl1.BorderStyle = System.Windows.Forms.BorderStyle.Fixed3D;
            this.DigitalIOCtrl1.Location = new System.Drawing.Point(9, 9);
            this.DigitalIOCtrl1.Name = "DigitalIOCtrl1";
            this.DigitalIOCtrl1.Size = new System.Drawing.Size(255, 343);
            this.DigitalIOCtrl1.TabIndex = 5;
            this.DigitalIOCtrl1.Visible = true;

            this.Controls.Add(DigitalIOCtrl1);
            this.Refresh();

            int numCtrls = this.Controls.Count;
        }

        private void close_btn_Click(object sender, EventArgs e)
        {
            this.Visible = false;
        }

    }
}

File: cableTester/cableTester/IO_frm.vb
Public Class IO_frm
    Private Sub hide_btn_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles hide_btn.Click
        Me.Hide()
    End Sub

    Public Sub addDIOCtrl(ByVal niPath As String, ByVal equipmentH As equipmentHandler.handler)
        If Not equipmentH.equipment Is Nothing Then
            For cntr = 0 To equipmentH.equipment.Length - 1
                If equipmentH.equipment(cntr).type = "DAQ" Then

                    ' create digital IO control from DAQ instrument
                    Me.DigitalIOCtrl1 = New dio_ctrl.DigitalIOCtrl(niPath, equipmentH.equipment(cntr).model, equipmentH.equipment(cntr).device)

                    Me.DigitalIOCtrl1.BorderStyle = System.Windows.Forms.BorderStyle.Fixed3D
                    Me.DigitalIOCtrl1.Location = New System.Drawing.Point(9, 9)
                    Me.DigitalIOCtrl1.Name = "DigitalIOCtrl1"
                    Me.DigitalIOCtrl1.Size = New System.Drawing.Size(255, 343)
                    Me.DigitalIOCtrl1.TabIndex = 5
                    Me.DigitalIOCtrl1.Visible = True

                    Me.Controls.Add(DigitalIOCtrl1)
                    Me.Refresh()

                    Dim numCtrls As Integer = Me.Controls.Count

                    Exit For
                End If
            Next
        End If
    End Sub

    Private Sub IO_frm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        DigitalIOCtrl1.Refresh()
    End Sub
End Class